org.jacoco.core/src/org/jacoco/core/matcher/Matcher.java

```diff
@@ -0,0 +1,16 @@
+/*******************************************************************************
+ * Copyright (c) 2009, 2017 Mountainminds GmbH & Co. KG and Contributors
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Jeffry Gaston - initial API and implementation
+ *
+ *******************************************************************************/
+package org.jacoco.core.matcher;
+
+public interface Matcher<T> {
+	boolean matches(final T s);
+}
```

Comment:
Also, when you do come back to this pull request, it'd be good to know your opinions about adding a dependency like Guava or apache-commons that already contains an interface like this, as opposed to redefining an interface here.

It happens to be slightly more convenient for Android if we avoid the dependency, although I imagine that whichever approach is best for Jacoco itself would be the one we'd want to actually do.